azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_UpdateMachineProperties.ps1

```diff
@@ -135,6 +135,60 @@ Function ProcessItemImpl($processor, $csvItem, $reportItem) {
     } 
     #End Code for Target Subscription
 
+    $targetDiskName = $csvItem.UPDATED_TARGET_DISK_NAME
+    if ([string]::IsNullOrEmpty($targetDiskName)) {
+        $processor.Logger.LogTrace("UPDATED_TARGET_DISK_NAME is not mentioned for: '$($sourceMachineName)'")
+    }
+    else {
+        $params.Add("TargetDiskName", $targetDiskName)
+    }
+
+    $diskMapping = @()
+    $paramsDisk1 = @{}
+    $osDiskId = $csvItem.OS_DISK_ID
+    $osDiskName = $csvItem.UPDATED_TARGET_OS_DISK_NAME
+
+    if ([string]::IsNullOrEmpty($osDiskId) -or [string]::IsNullOrEmpty($osDiskName)) {
+        $processor.Logger.LogTrace("OS_DISK_ID or UPDATED_TARGET_OS_DISK_NAME is not mentioned for: '$($sourceMachineName)'")
+        $reportItem.AdditionalInformation = "OS_DISK_ID or UPDATED_TARGET_OS_DISK_NAME is not mentioned for: '$($sourceMachineName)'"
+    }
+    else {
+        $paramsDisk1.Add("DiskId", $osDiskId)
+        $paramsDisk1.Add("IsOSDisk", $true)
+        $paramsDisk1.Add("TargetDiskName", $osDiskName)
+        $diskMapping+= $paramsDisk1
+    }
+
+    $paramsDisk2 = @{}
+    $dataDisk1Id = $csvItem.DATA_DISK1_ID
+    $dataDisk1Name = $csvItem.UPDATED_TARGET_DATA_DISK1_NAME
+    if ([string]::IsNullOrEmpty($dataDisk1Id) -or [string]::IsNullOrEmpty($dataDisk1Name)) {
+        $processor.Logger.LogTrace("DATA_DISK1_ID or UPDATED_TARGET_DATA_DISK1_NAME is not mentioned for: '$($sourceMachineName)'")
+        $reportItem.AdditionalInformation = "DATA_DISK1_ID or UPDATED_TARGET_DATA_DISK1_NAME is not mentioned for: '$($sourceMachineName)'"
+    }
+    else {
+        $paramsDisk2.Add("DiskId", $dataDisk1Id)
+        $paramsDisk2.Add("IsOSDisk", $false)
+        $paramsDisk2.Add("TargetDiskName", $dataDisk1Name)
+        $diskMapping+= $paramsDisk2
+    }
+
+    $paramsDisk3 = @{}
+    $dataDisk2Id = $csvItem.DATA_DISK2_ID
+    $dataDisk2Name = $csvItem.UPDATED_TARGET_DATA_DISK2_NAME
+    if ([string]::IsNullOrEmpty($dataDisk2Id) -or [string]::IsNullOrEmpty($dataDisk2Name)) {
```

Comment:
[string]::IsNullOrEmpty($dataDisk2Name) should be present inside else statement

Its possible that user only wants to change the OS disk and not the name.